waspc/cli/src/Wasp/Cli/Command/Uninstall.hs

```diff
@@ -0,0 +1,63 @@
+module Wasp.Cli.Command.Uninstall
+  ( uninstall,
+  )
+where
+
+import Control.Monad (when)
+import Control.Monad.IO.Class (liftIO)
+import StrongPath (Dir', File', Path', Rel, fromAbsDir, fromAbsFile, reldir, relfile, (</>))
+import System.Exit (die)
+import Wasp.Cli.Command (Command)
+import Wasp.Cli.Command.Message (cliSendMessageC)
+import Wasp.Cli.Command.Telemetry.Common (getWaspCacheDirPath)
+import Wasp.Cli.Command.Utils.File (deleteDirectoryIfExists, deleteFileIfExists, getUserCacheDirPath)
+import Wasp.Cli.Command.Utils.User (HomeDir, getHomeDir)
+import qualified Wasp.Message as Msg
+
+-- | Removes Wasp CLI from the system.
+-- It removes the follwing:
+-- {home}/.local/share/wasp-lang
+-- {home}/.local/bin/wasp
+-- {home}/.cache/wasp
+uninstall :: Command ()
+uninstall = do
+  cliSendMessageC $ Msg.Info "Uninstalling Wasp ..."
+
+  liftIO removeWaspFiles
+
+  cliSendMessageC $ Msg.Success "Uninstalled Wasp"
+
+removeWaspFiles :: IO ()
+removeWaspFiles = do
+  homeDir <- getHomeDir
+  userCacheDir <- getUserCacheDirPath
+
+  let waspInstallationDir' = fromAbsDir $ homeDir </> waspInstallationDir
+      waspBinFile' = fromAbsFile $ homeDir </> waspBinFile
+      waspCacheDir = fromAbsDir $ getWaspCacheDirPath userCacheDir
+
+  putStr $
+    unlines
+      [ "We will remove the following directories:",
+        "  " ++ waspInstallationDir',
+        "  " ++ waspCacheDir,
+        "",
+        "We will also remove the following files:",
+        "  " ++ waspBinFile',
+        "",
+        "Are you sure you want to continue? [y/N]"
+      ]
+
+  answer <- getLine
+
+  when (answer /= "y") $ die "Aborted."
```

Comment:
Are we sure we don't want to be case insensitive? I don't know what's the usual convention. Btw, the message says `y/N`, is the different capitalization on purpose?

Minor nitpick: I'd also drop the spacing  on lines 50 and 52.

Reply:
I think it's fine to require exact char `y`, we are deleting files after all. 

Also, I've seen the `y/N` used in some other places which indicates that `N` is the default choice.

<img width="607" alt="Screenshot 2023-01-25 at 09 29 46" src="https://user-images.githubusercontent.com/2223680/214515148-e50727f6-bed8-408b-bea2-0dc1a1bfbc0f.png">